pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -199,8 +201,15 @@ protected BrokerResponse handleRequest(long requestId, String query, SqlNodeAndO
     }
 
     DispatchableSubPlan dispatchableSubPlan = queryPlanResult.getQueryPlan();
-    Set<String> tableNames = queryPlanResult.getTableNames();
 
+    Set<QueryServerInstance> servers = dispatchableSubPlan.getQueryStageList().stream()
+        .map(DispatchablePlanFragment::getServerInstances)
+        .reduce(new HashSet<QueryServerInstance>(), (all, some) -> {
+          all.addAll(some);
+          return all;
+        });
```

Comment:
This is correct, but unnecessarily complex to read. What about

What about something like:
```
dispatchableSubPlan.getQueryStageList().stream()
        .map(DispatchablePlanFragment::getServerInstances)
        .reduce(new HashSet<>(), Sets::union)
        .size();
```

Other options (maybe a bit more expensive, but easier to read):
```
dispatchableSubPlan.getQueryStageList().stream()
        .flatMap(fragment -> fragment.getServerInstances().stream())
        .distinct()
        .count();
```
or
```
dispatchableSubPlan.getQueryStageList().stream()
        .flatMap(fragment -> fragment.getServerInstances().stream())
        .collect(Collectors.toSet())
        .size();
```

Reply:
I'm not a big fan of functional API (don't really know what's happening behind the scene, and higher overhead), so I'd suggest just using the old fashioned way of creating a set and calling `addAll()`

Reply:
+1. I don't find this coding pattern readable.

Reply:
I think any of the 3 versions I suggested is easy to read (probably more than a for loop)

Reply:
I'm gonna betray Gonzalo here 😇

I'm more into using a conventional loop with addAll, too. Using streams here feels a bit overkill for what the code intends to do. Going through the whole array anyway, so I don't see the point of relying into streams.

Actually, that was my first draft and changed it afterwards trying to show off. It obviously didn't go as intended, so conventional loop it is: [de63ddeead](https://github.com/apache/pinot/pull/14806/commits/de63ddeeadbebdfe88910179140046bab66bfcf2)